### ISSUE TITLE
[Linux] System.DirectoryServices.Protocols is not supported on this platform

### ISSUE THREAD

**Issue by Lemaus123** [dotnet/runtime#77022]
I have error of System.DirectoryServices.Protocols is not supported on this plartform while I'm using 5.18.19-3-MANJARO (64-bit), with the 6.0.9 runtime

The exception is thrown while executing the following statement:
```cs
var connection = new LdapConnection(domain);
```

I tried both versions 6.0.1 and 7.0.0-rc.2.22472.3, and the issue is occurring every time.  

I use .NET6 for this project. According to the research I've done, this package should be cross-platform on both Mac and Linux for a few years now, am I missing something to be able to use it?

**Comment by vcsjones** [comment]
> I have error of System.DirectoryServices.Protocols is not supported on this plartform

What is the exact error you are getting, with a stack trace?

> this package should be cross-platform on both Mac and Linux for a few years now, am I missing something to be able to use it?

You do need libldap-2.4.so.2 available on your system. Do note that if your distribution comes with a newer version of libldap like 2.5 or 2.6, there is a known issue loading those. https://github.com/dotnet/runtime/issues/69456

**Comment by Lemaus123** [comment]
Hi there,

Here is the stack trace
```cs
System.PlatformNotSupportedException: System.DirectoryServices.Protocols is not supported on this platform.
   at System.DirectoryServices.Protocols.DirectoryConnection..ctor()
   at System.DirectoryServices.Protocols.LdapConnection..ctor(String server)
   at CRM_API.Utilities.GetADUsers() in /home/lopb/Desktop/CRM-API/Utilities.cs:line 16
   at CRM_API.Controllers.UsersController.Get() in /home/lopb/Desktop/CRM-API/Controllers/UsersController.cs:line 16
   at lambda_method3(Closure , Object , Object[] )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
```

Both libldap and lib32-libldap 2.6.3-1 are already installed on my system.
Those libraries are required for my project since I use postgresql as database.

When I compare the stack trace of the issue you linked and mine, they don't seem to be related.

**Comment by vcsjones** [comment]
@Lemaus123 what is the output of `dotnet --info` on your system, and how are you consuming the package, and what is the `<TargetFramework>` of your application? It looks like you're getting the default "I don't know what target platform you are on so it's not supported"

https://github.com/dotnet/runtime/blob/888ad2bab1fc7ae25b4e27cf31dbd817eeb306d5/src/libraries/System.DirectoryServices.Protocols/src/System.DirectoryServices.Protocols.csproj#L18

**Comment by joperezr** [comment]
Thanks for logging the issue @Lemaus123 and thanks for the replies @vcsjones, indeed this looks like a case where the package is either not consumed correctly or the app is not being deployed correctly, as @Lemaus123 is getting the RIDless configuration as opposed to getting the Linux one. @Lemaus123 if you could share the code inside your .csproj as well as the instructions on how you are deploying and running your app it would be helpful in figuring out what is going on here.

**Comment by Lemaus123** [comment]
Here is the output of dotnet --info
```
.NET SDK (reflecting any global.json):
 Version:   6.0.109
 Commit:    58a93139d8

Runtime Environment:
 OS Name:     manjaro
 OS Version:  
 OS Platform: Linux
 RID:         arch-x64
 Base Path:   /usr/share/dotnet/sdk/6.0.109/

global.json file:
  Not found

Host:
  Version:      6.0.9
  Architecture: x64
  Commit:       163a63591c

.NET SDKs installed:
  6.0.109 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.9 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.9 [/usr/share/dotnet/shared/Microsoft.NETCore.App]


```

Here is my .csproj

```cs
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <Nullable>disable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
    <RootNamespace>CRM_API</RootNamespace>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="AutoMapper" Version="12.0.0" />
    <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="12.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="6.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="6.0.10" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="6.0.10" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.10" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="6.0.10">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="6.0.10" />
    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
    <PackageReference Include="Npgsql" Version="6.0.7" />
    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="6.0.7" />
    <PackageReference Include="Swashbuckle.AspNetCore" Version="6.4.0" />
    <PackageReference Include="System.DirectoryServices.Protocols" Version="6.0.1" />
  </ItemGroup>

</Project>

```
I'm using Ryder so i'm not sure what to answer about how i'm running my app since its all automatic.

My coworker use it on the same project on his Windows OS and the package work very well, so the way the package is consumed should not be the problem.

**Comment by wzchua** [comment]
Related https://github.com/dotnet/runtime/issues/62554 ?

**Comment by Lemaus123** [comment]
With all your help I figured out a way to make this work for now. By adding 
```cs
    <RuntimeIdentifier>linux-x64</RuntimeIdentifier>
```
in my .csproj the platform was correctly identified. 
I had an error of **Unable to load shared library 'libldap-2.4.so.2' or one of its dependencies.**. (because the version is hard coded in .net runtime). So I downloaded the libldap24 from the AUR and everything started to work well.
